### PR TITLE
add concept difficulties to all K-5 levels

### PR DESCRIPTION
**Pull request by Hamms**
depends on #7767.

To generate these changes, I first grabbed the existing tags from the reporting database:

```
 SELECT scripts.name, stage_position, level_position, sequencing, debugging, repeat_loops, repeat_until_while, for_loops, events, variables, functions, functions_with_params FROM csf_concept_tags JOIN scripts ON csf_concept_tags.script_id = scripts.id;
```

I then dumped that data into a CSV, which I loaded into my local db using the following script:

``` ruby
require_relative('../config/environment')
require 'cdo/properties'

def main()
  CSV.foreach("/tmp/level_concept_difficulties.csv") do |row|
    level = Script.where(name: row[0]).first.stages.where(position: row[1]).first.script_levels.where(position: row[2]).first.level
    concepts = row.slice(3..row.length).map do |val|
      val == "NULL" ? nil : val
    end
    lcd = Hash[LevelConceptDifficulty::CONCEPTS.zip(concepts)]
    lcd[:level] = level
    LevelConceptDifficulty.create(lcd)
    level.write_custom_level_file
  end
end

main()

```

I then went and did this all again from the [spreadsheet](https://docs.google.com/spreadsheets/d/1IUcvX50-olC2ud4gIaqYpKXwRnaI2bHFs1uJUA4HgrA/edit#gid=980286757) so we could get up-to-date data. I downloaded the csvs and loaded them in using this script:

``` ruby
require_relative('../config/environment')
require 'cdo/properties'

def main()
  %w(20-hour course1 course2 course3 course4).each do |course|
    CSV.foreach("/home/elijah/code-dot-org/concept_tags/#{course}.csv", headers: true,  header_converters: :symbol) do |row|
      concept_difficulties = row.to_hash.compact.delete_if do |key, val|
        !LevelConceptDifficulty::CONCEPTS.include?(key.to_s)
      end

      level = Script.where(name: course).first
          .stages.where(position: row[:stage_position]).first
          .script_levels.where(position: row[:level_position]).first.level

      lcd = LevelConceptDifficulty.find_or_create_by(level: level)
      lcd.update(concept_difficulties)
      level.write_custom_level_file
    end
  end
end

main()

```

### CODE REVIEW

File: dashboard/config/scripts/levels/2-3 Artist 1 new.level [hunk 1]
```diff
@@ -38,7 +38,19 @@
...
+    "sequencing": 3,
+    "debugging": null,
+    "repeat_loops": null,
+    "repeat_until_while": null,
+    "for_loops": null,
+    "events": null,
+    "variables": null,
+    "functions": null,
+    "functions_with_params": null,
+    "conditionals": null
```
Comment (ashercodeorg) [inline dashboard/config/scripts/levels/2-3 Artist 1 new.level]: Is it possible to make this more succinct by omitting the null values?

**Comment by ashercodeorg** [comment]
Let me confirm with Ryan (offline) that the spreadsheet (until the DB, the source of truth) hasn't changed since I last updated the reporting csf_concept_tags table.

File: dashboard/config/scripts/levels/2-3 Artist 1 new.level (hunk 1 above)
Comment (Hamms) [inline dashboard/config/scripts/levels/2-3 Artist 1 new.level]: Probably, this is just the way `Level.serializable_hash(:include => :level_concept_difficulty)` works by default. I'll see if it (and more importantly `assign_nested_attributes_for_one_to_one_association`) are happy with leaving out nulls

Comment (Hamms) [inline dashboard/config/scripts/levels/2-3 Artist 1 new.level]: [done](https://github.com/code-dot-org/code-dot-org/pull/7767/commits/53b055bc64b093b255f5af59c0be296c8a758f6d)

Comment (ashercodeorg) [inline dashboard/config/scripts/levels/2-3 Artist 1 new.level]: Thanks!

**Comment by Hamms** [comment]
both this and #7767 are ready for final looks

**Comment by ashercodeorg** [comment]
LGTM.
1. You should update the PR summary to indicate the tags were ripped from the spreadsheet (give a link) rather than the reporting database so as to grab the latest changes.
2. We should deprecate the spreadsheet as quickly as possible after this goes in.
3. We should deprecate the csf_concept_tags table as quickly as possible after this goes in.

And I confirmed things match the reporting database via spot-checking a few random levels.

**Pull request merged**